Code/Editor/CryEdit.cpp

```diff
@@ -3959,32 +3960,64 @@ void CCryEditApp::OpenLUAEditor(const char* files)
         }
     }
 
-    AZ::IO::FixedMaxPathString engineRoot = AZ::Utils::GetEnginePath();
-    AZ_Assert(!engineRoot.empty(), "Unable to query Engine Path");
+    const char* engineRoot = nullptr;
+    AzFramework::ApplicationRequests::Bus::BroadcastResult(engineRoot, &AzFramework::ApplicationRequests::GetEngineRoot);
+    AZ_Assert(engineRoot != nullptr, "Unable to communicate to AzFramework::ApplicationRequests::Bus");
+    AZStd::string externalProcess = gSettings.textEditorForScript.toUtf8().data();
+    AZStd::string process;
+    AZStd::string processArgs;
 
+    AZStd::string projectPath;
+    if (auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
+    {
+        settingsRegistry->Get(projectPath, AZ::SettingsRegistryMergeUtils::FilePathKey_ProjectPath);        
+    }
+    
+    // only launch IDEA for Lua Script
+    if (externalProcess.ends_with("idea64.exe") || externalProcess.ends_with("idea.exe") || externalProcess.ends_with("idea.sh"))
+    {
+        process = "\"" + externalProcess + "\"";
+        const auto projectName = AZ::Utils::GetProjectName();
+        AZStd::string projectRoot = AZStd::string::format("%s/%s", engineRoot, projectName.c_str());
+        if (fileName.empty())
+        {
+            processArgs = projectRoot;
+        }
+        else
+        {
+            processArgs = "\"" + fileName + "\"";
+        }
+        
+        //startup idea
+        StartProcessDetached(process.c_str(), processArgs.c_str());
+
+        // This parameter determines Lua IDE Running in the background.
+        args += "\"-runbackground=lua\"";
+    }
+   
     AZStd::string_view exePath;
     AZ::ComponentApplicationBus::BroadcastResult(exePath, &AZ::ComponentApplicationRequests::GetExecutableFolder);
 
-#if defined(AZ_PLATFORM_LINUX)
-    // On Linux platforms, launching a process is not done through a shell and its arguments are passed in
-    // separately. There is no need to wrap the process path in case of spaces in the path
-    constexpr const char* argumentQuoteString = "";
-#else
-    constexpr const char* argumentQuoteString = "\"";
-#endif    
-
-    AZStd::string process = AZStd::string::format("%s%.*s" AZ_CORRECT_FILESYSTEM_SEPARATOR_STRING "LuaIDE"
+    process = AZStd::string::format(
+        "\"%.*s" AZ_CORRECT_FILESYSTEM_SEPARATOR_STRING "LuaIDE"
 #if defined(AZ_PLATFORM_WINDOWS)
         ".exe"
 #endif
-        "%s", argumentQuoteString, aznumeric_cast<int>(exePath.size()), exePath.data(), argumentQuoteString);
-
-    AZStd::string processArgs = AZStd::string::format("%s -engine-path \"%s\"", args.c_str(), engineRoot.c_str());
+        "\"",
+        aznumeric_cast<int>(exePath.size()), exePath.data());
+
+    //setting startup parameter
+    processArgs = AZStd::string::format("%s", args.c_str());
+    processArgs += "\"" + AZStd::string::format("--project-path=%s\"", projectPath.c_str());
+    processArgs += "\"" + AZStd::string::format("-engine-path=%s\"", engineRoot);
+   
```

Comment:
Based on suggestion to use a vector of strings for arg, and the need to double quote should come after the equal sign:

```suggestion
    args.emplace_back(AZStd::string::format(R"(--project-path="%s")", projectPath.c_str()));
    args.emplace_back(AZStd::string::format(R"(-engine-path="%s")", engineRoot)));

    AZ::StringFunc::Join(processArgs, args.begin(), args.end(), " ");  
```